lib/Wasmtime.hs

```diff
@@ -1318,9 +1336,11 @@ peekTableVal :: Ptr C'wasmtime_val_t -> IO TableValue
 peekTableVal val_ptr = do
   k <- peek kind_ptr
   if
-      | k == c'WASMTIME_FUNCREF ->
-          FuncRefValue . MkFunc
-            <$> peek (castPtr of_ptr :: Ptr C'wasmtime_func_t)
+      | k == c'WASMTIME_FUNCREF -> do
+          func <- MkFunc <$> mallocForeignPtr
```

Comment:
and the free to this malloc is in the finalizer of the store, correct? perhaps we should leave a comment here as well.

Reply:
Yeah `mallocForeignPtr` allocates some memory and creates a `ForeignPtr` which frees that memory in its finalizer.